packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -162,10 +180,16 @@ const handleOriginRequest = async ({
   const normalisedS3DomainName = normaliseS3OriginDomain(s3Origin);
   const hasFallback = hasFallbackForUri(uri, prerenderManifest);
   const { now, log } = perfLogger(manifest.logLambdaExecutionTimes);
+  const isPreviewRequest = checkPreviewRequest(request);
```

Comment:
Could we check for "preview request" only if previewMode is enabled? There is a perf impact on looking for the preview mode cookie on every req that hopefully we can do only if necessary.

Reply:
I think I'm missing a trick - I thought the only way to determine preview mode _is_ due to the presence of the cookies. There isn't a build config or any static info that we can check I think.

Reply:
I think you're right, seems like the only way to determine this is looking if the cookies exist :/ Next.js is generating in the `prerender-manifest.json` a "preview" object that will always be there on each build.